container/container.go

```diff
@@ -46,6 +46,8 @@ const configFileName = "config.v2.json"
 var (
 	errInvalidEndpoint = fmt.Errorf("invalid endpoint while building port map info")
 	errInvalidNetwork  = fmt.Errorf("invalid network settings while building port map info")
+	// ErrDetached implies container is detached
+	ErrDetached = fmt.Errorf("detached from container")
```

Comment:
Create type for `detachErr` with an interface `IsDetached`

```go
type AttachError interface {
    IsDetached() bool
}

type detachErr struct {
    err error
}
func(detachErr) IsDetached() bool {
    return true
}

func(e *detachErr) Error() string {
    return e.err
}
```

Then in daemon we do
```go
if e, ok := err.(container.AttachErr); ok {
    if e.IsDetached() {
        // log detach event
    }
}
```